### PR TITLE
feat(typography): add code-block component page

### PR DESCRIPTION
**Pull request by jzempel**
## Description

Adds a new `Typography / Code block` component page.

## Detail

@steue I winged it on the examples and several copy modifications. Please review with a critical eye.

## Checklist

- [x] :ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)
- [x] :black_nib: copy updates are approved (add the content strategist as a reviewer)
- [x] :link: considered opportunities for adding cross-reference URLs (grep for keywords)
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [ ] :zap: ~audited via [web.dev](https://web.dev/measure/) for performance and SEO~
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: src/pages/components/code-block.mdx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import Default from '../../examples/typography/code-block/Default.tsx';
+import DefaultCode from '!!raw-loader!../../examples/typography/code-block/Default.tsx';
+
+<CodeExample code={DefaultCode}>
+  <Default />
+</CodeExample>
+
+### Light mode
+
+By default, Code block is dark. Shed some light on it using `isLight`.
```
Comment (allisonacs) [inline src/pages/components/code-block.mdx:45]: Capitalization of "Code" but not "block" feels weird here. Either the whole thing is a proper noun ("Code Block") or it's not ("code block").

File: src/pages/components/code-block.mdx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import Light from '../../examples/typography/code-block/Light.tsx';
+import LightCode from '!!raw-loader!../../examples/typography/code-block/Light.tsx';
+
+<CodeExample code={LightCode}>
+  <Light />
+</CodeExample>
+
+### Language
+
+Use the `language` prop to determine the syntax highlighting language. Default is `tsx`.
```
Comment (allisonacs) [inline src/pages/components/code-block.mdx:56]: "to determine" here sounds like the language prop performs some evaluation to learn what language is displayed. 

Perhaps:

> Use the `language` proper to specify the syntax highlighting language.

File: src/pages/components/code-block.mdx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import Language from '../../examples/typography/code-block/Language.tsx';
+import LanguageCode from '!!raw-loader!../../examples/typography/code-block/Language.tsx';
+
+<CodeExample code={LanguageCode}>
+  <Language />
+</CodeExample>
+
+### Line numbers
+
+Code block supports line numbers, improving the ability to reference lines of code.
```
Comment (allisonacs) [inline src/pages/components/code-block.mdx:67]: This is a bit of a mouthful. Perhaps:

> Line numbers make it easier to reference specific lines of code. You can turn this on with the `isNumbered` prop.

**Review by allisonacs — CHANGES_REQUESTED** [review]
Couple of content issues. I've made some suggestions, but @vsrixyz should chime in too.

File: src/pages/components/code-block.mdx (hunk 1 above)
Comment (jzempel) [reply src/pages/components/code-block.mdx:45]: @allisonacs this is the capitalization convention for multi-word components and would be a consistency update across many pages of the website. See
- https://garden.zendesk.com/components/button-group
- https://garden.zendesk.com/components/icon-button
- https://garden.zendesk.com/components/multi-thumb-range
- https://garden.zendesk.com/components/tooltip-modal
- etc

File: src/pages/components/code-block.mdx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+import LanguageCode from '!!raw-loader!../../examples/typography/code-block/Language.tsx';
+
+<CodeExample code={LanguageCode}>
+  <Language />
+</CodeExample>
+
+### Line numbers
+
+Line numbers make it easier to reference specific lines of code. You can turn
+this on with the `isNumbered` prop.
```
Comment (steue) [inline src/pages/components/code-block.mdx:68]: I was just about to mention this, I'm glad you dropped it in!

**Review by steue — APPROVED** [review]
this all looks great. the examples are feeling more illustrative!

**Review by austingreendev — APPROVED** [review]

**Review by allisonacs — APPROVED** [review]

**Comment by jzempel** [comment]
@vsrixyz even though this is merged, I'm happy to take follow-on comments for copy amendments. 🥂

**Comment by vsrixyz** [comment]
> @vsrixyz even though this is merged, I'm happy to take follow-on comments for copy amendments. 🥂

Took another look at this and it looks good to me.

**Pull request merged**